File: 백준/Silver/10828. 스택/스택.py
import sys
input= sys.stdin.readline

stack=[]

n=int(input())
for _ in range(n):
    command=input().split()
    if command[0]=='push':
        stack.append(command[1])
    elif command[0]=='pop':
        if not len(stack):
            print(-1)
        else:
            print(stack.pop())
    elif command[0]=='size':
        print(len(stack))
    elif command[0]=='empty':
        if not len(stack):
            print(1)
        else: print(0)
    else:
        if not len(stack):
            print(-1)
        else: print(stack[-1])

File: 백준/Silver/10828. 스택/스택.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;

public class Main {
    public static void main(String[] args) throws IOException {
        Stack<Integer> stack = new Stack<>();
        BufferedReader br =new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        while(N-- > 0){
            String[] input = br.readLine().split(" ");
            switch (input[0]){
                case "push":
                    stack.push(Integer.parseInt(input[1]));
                    break;
                case "top" :
                    if(stack.empty()){
                        System.out.println(-1);
                    }else{
                        System.out.println(stack.peek());
                    }
                    break;
                case "size" :
                    System.out.println(stack.size());
                    break;
                case "empty" :
                    if(stack.empty()){
                        System.out.println(1);
                    }else{
                        System.out.println(0);
                    }
                    break;
                case "pop" :
                    if(stack.empty()){
                        System.out.println(-1);
                    }else{
                        System.out.println(stack.peek());
                        stack.pop();
                    }
                    break;
            }
        }
    }
}